eng/pipelines/scripts/generate-project-list.ps1

```diff
@@ -23,18 +25,55 @@ if ($env:ADDITIONALMODULESJSON -and $env:ADDITIONALMODULESJSON -notlike '*Additi
 # which means this is running as part of the pullrequest pipeline and the project list needs to
 # be figured out from the packageInfo files.
 if ($projectList.Length -eq 0 -and $ENV:PACKAGEINFODIR) {
-  [array]$packageInfoFiles = Get-ChildItem -Path $ENV:PACKAGEINFODIR "*.json"
-  foreach($packageInfoFile in $packageInfoFiles) {
-    $packageInfoJson = Get-Content $packageInfoFile -Raw
-    $packageInfo = ConvertFrom-Json $packageInfoJson
-    $fullArtifactName = "$($packageInfo.Group):$($packageInfo.ArtifactName)"
-    $projectList += $fullArtifactName
-    $artifactsList += $fullArtifactName
-    # The AdditionalValidationPackages are stored as <group>:<artifact>
-    foreach($additionalModule in $packageInfo.AdditionalValidationPackages)
-    {
-      $projectList += $additionalModule
-      $additionalModulesList += $additionalModule
+
+  # This is the case where this is being called as part of the set of test matrix runs.
+  # The ArtifactPackageNames environment variable will be set if this is being called
+  # as one of the test matrix runs. In this case, the project and additional modules lists
+  # need to be filtered by the ArtifactPackageNames otherwise there will be artifacts on
+  # the maven command line, for a matrix, that don't belong to the matrx if the PR has
+  # changes to multiple libraries that have different test matrices.
+  if (-not [string]::IsNullOrEmpty($ENV:ARTIFACTPACKAGENAMES)) {
+    Write-Host "ArtifactPackageNames is set to: $($ENV:ARTIFACTPACKAGENAMES)"
+    # The ArtifactPackageNames is a comma separated list
+    foreach ($artifactPackageName in $ENV:ARTIFACTPACKAGENAMES.Split(',')) {
+      # There should only be 1 PackageInfo file for each ArtifactPackageName.
+      [array]$packageInfoFiles = Get-ChildItem -Path $ENV:PACKAGEINFODIR "$($artifactPackageName).json"
+      if ($packageInfoFiles) {
+        if ($packageInfoFiles.Length -gt 1) {
+          LogWarning "Multiple PackageInfo files found for $artifactPackageName"
```

Comment:
Should this be an error?

Reply:
I'm keeping the warning here but, you're right, I should just use the first one though this isn't a case that should ever happen. The next commit will log the warning and just use the first one returned.